src/core/comm/alltoall_thread_local.cc

```diff
@@ -0,0 +1,105 @@
+/* Copyright 2022 NVIDIA Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+#include <assert.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include "coll.h"
+#include "legion.h"
+
+namespace legate {
+namespace comm {
+namespace coll {
+
+using namespace Legion;
+extern Logger log_coll;
+
+int collAlltoallLocal(const void* sendbuf,
+                      int sendcount,
+                      CollDataType sendtype,
+                      void* recvbuf,
+                      int recvcount,
+                      CollDataType recvtype,
+                      CollComm global_comm)
+{
+  int res;
+
+  assert(recvcount == sendcount);
+  assert(sendtype == recvtype);
+
+  int total_size = global_comm->global_comm_size;
+
+  int sendtype_extent = collGetDtypeSize(sendtype);
+  int recvtype_extent = collGetDtypeSize(recvtype);
+
+  int global_rank = global_comm->global_rank;
+
+  void* sendbuf_tmp = NULL;
```

Comment:
I've already seen this pattern three times. wouldn't it be better to extract this into a function?

Reply:
I cleaned it up https://github.com/eddy16112/legate.core/blob/coll/src/core/comm/allgather_thread_local.cc#L40-L54
I just feel like it is not a good way to put this 10 lines of code into a function, because all these 5 parameters are used later. If I put them into a function, then we will need to give 7-8 inputs and 5 outputs.